e2e/multi-svc-app/multi_svc_app_test.go

```diff
@@ -338,6 +352,56 @@ var _ = Describe("Multiple Service App", func() {
 			Expect(string(bodyBytes)).To(Equal("back-end-service"))
 		})
 
+		It("should be able to send udp message", func() {
+			// The front-end service has an NLB listener on port 8080 over UDP.
+			svcName := "front-end"
+			svc, svcShowErr := cli.SvcShow(&client.SvcShowRequest{
+				AppName: appName,
+				Name:    svcName,
+			})
+			Expect(svcShowErr).NotTo(HaveOccurred())
+			Expect(len(svc.Routes)).To(Equal(1))
+
+			route := svc.Routes[0]
+
+			// route.URL is of the form `example-alb.elb.us-west-2.amazonaws.com or example-nlb.elb.us-west-2.amazonaws.com, so we split to retrieve just one valid url`
+			routeURLs := strings.Split(route.URL, " ")
+			Expect(len(routeURLs)).To(BeNumerically(">", 1))
+
+			routeURL = routeURLs[0]
+
+			Expect(route.Environment).To(Equal("test"))
+
+			conn, dialErr := net.Dial("udp", routeURL)
+			Expect(dialErr).NotTo(HaveOccurred())
+
+			// Send message 5 times in case UDP packets are dropped.
+			testStr := "test message"
+			for i := 0; i < 5; i++ {
+				conn.Write([]byte(testStr))
+				time.Sleep(time.Second)
+			}
+
+			// Retrieve logs to check if UDP traffic was received.
+			var svcLogs []client.SvcLogsOutput
+			var svcLogsErr error
+			Eventually(func() ([]client.SvcLogsOutput, error) {
+				svcLogs, svcLogsErr = cli.SvcLogs(&client.SvcLogsRequest{
+					AppName: appName,
+					Name:    svcName,
+					EnvName: "test",
+					Since:   "1h",
+				})
+				return svcLogs, svcLogsErr
+			}, "60s", "10s").ShouldNot(BeEmpty())
+
+			var svcLogMessages []string
+			for _, logLine := range svcLogs {
+				svcLogMessages = append(svcLogMessages, logLine.Message)
+			}
+			Expect(svcLogMessages).To(ContainElement(ContainSubstring("Received UDP message: test message")))
```

Comment:
should we parse the messages in the `Eventually()` above?

Reply:
I see your point here. I guess the concern is that there are log entries that aren't from the UDP traffic but cause the `Eventually()` to return, and then we miss the UDP packet if for some reason it's slow to arrive or be logged. I find this possibility insanely unlikely given the timescale we're checking on, but I guess it's still possible 👍